plugin/references.py

```diff
@@ -113,17 +130,24 @@ def _handle_response(
         force_group: bool,
         fallback: bool,
         group: int,
+        show_in: Optional[ShowInArgument],
+        event: Optional[dict],
         position: int,
         response: Optional[List[Location]]
     ) -> None:
-        if response:
-            if userprefs().show_references_in_quick_panel:
-                self._show_references_in_quick_panel(
-                    word, session, response, side_by_side, force_group, group, position)
-            else:
-                self._show_references_in_output_panel(word, session, response)
-        else:
+        if not response:
             self._handle_no_results(fallback, side_by_side)
+            return
+        modifier_keys = (event or {}).get('modifier_keys', {})
+        show_in_quick_panel = show_in == 'quick_panel' or show_in is None and userprefs().show_references_in_quick_panel
+        if modifier_keys.get('shift'):
+            show_in_quick_panel = not show_in_quick_panel
```

Comment:
I've realized that I should only be handling shift when "show_in" is not specified.